src/sinks/util/service.rs

```diff
@@ -206,6 +212,17 @@ impl<T: ConcurrencyOption> TowerRequestConfig<T> {
             adaptive_concurrency: self.adaptive_concurrency,
         }
     }
+
+    fn concurrency(&self) -> &T {
+        match (self.concurrency.is_none(), self.in_flight_limit.is_none()) {
+            (_, true) => &self.concurrency,
+            (true, false) => &self.in_flight_limit,
+            (false, false) => {
+                warn!("`in_flight_limit` option has been renamed to `concurrency`. Ignoring `in_flight_limit` and using `concurrency` option");
+                &self.concurrency
+            }
+        }
```

Comment:
I'd prefer this logic be rewritten in terms of either `is_some` or the actual values (ie `match (&self.concurrency, &self.in_flight_limit)`) I had to read this a couple times before the `true`/`false` values made sense.